test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -885,8 +888,25 @@ public static String randomGeohash(int minPrecision, int maxPrecision) {
         return geohashGenerator.ofStringLength(random(), minPrecision, maxPrecision);
     }
 
-    public static String randomTestTransport() {
-        return randomBoolean() ? NioTransportPlugin.NIO_TRANSPORT_NAME : MockTcpTransportPlugin.MOCK_TCP_TRANSPORT_NAME;
+    private static final SetOnce<Boolean> useNio = new SetOnce<>();
```

Comment:
When I did that, I was getting a bunch of initialization exceptions on compilation. I assume there was some static initialization issue. I was doing this:

```
private static final SetOnce<Boolean> useNio = randomBoolean();
```